base/CMakeLists.txt

```diff
@@ -81,3 +81,25 @@ therock_cmake_subproject_glob_c_sources(rocm-half
     include
 )
 therock_cmake_subproject_activate(rocm-half)
+
+
+################################################################################
+# Artifacts
+################################################################################
+
+therock_provide_artifact(base
+  TARGET_NEUTRAL
+  DESCRIPTOR artifact.toml
+  COMPONENTS
+    run
+    dev
+    dbg
+    doc
+    test
+  SUBPROJECT_DEPS
+    rocm-cmake
+    rocm-core
+    rocm_smi_lib
```

Comment:
For Windows support (https://github.com/nod-ai/TheRock/issues/36), it would help to make some of these deps optional. For example, I don't expect rocm_smi_lib (or the successor project, https://github.com/ROCm/amdsmi) to compile on Windows.

Reply:
Yeah, we probably need some loose cmake to make some of them conditional as something like:

```
set(optional_artifact_deps)
if(NOT WIN32)
  list(APPEND optional_artifact_deps rocm_smi_lib)
endif()

...
  SUBPROJECT_DEPS
    ${optional_artifact_deps}
```

Could try to make it more automagical, but for this stuff, I generally prefer the readability of loose branchiness.